app/services/discussion_service.rb

```diff
@@ -40,22 +40,14 @@ def self.update(discussion:, params:, actor:)
     end
 
     return false unless discussion.valid?
-
-    update_search_vector = discussion.title_changed? || discussion.description_changed?
-
-    event = true
-    if discussion.title_changed?
-      event = Events::DiscussionTitleEdited.publish!(discussion, actor)
-    end
-
-    if discussion.description_changed?
-      event = Events::DiscussionDescriptionEdited.publish!(discussion, actor)
-    end
-
+    return discussion if discussion.changed == ['uses_markdown']
```

Comment:
Hey @robguthrie, I want to name the rules around what we return from services for update (and in general, really)

I think I prefer this:

- true (or nil) when it's successful and there is no event
- the event, if one has been generated
- raise an error if it fails (we want to know about it, no silent failures)
OR
- return false if it fails (this way we don't have to rescue in the existing discussions controller)

Does that sound reasonable?
Then this turns into
```
return false unless discussion.valid?
return true if discussion.changed.reject { |c| c == 'uses_markdown' }.empty?
```